src/System.CommandLine/Binding/ArgumentConversionResult.cs

```diff
@@ -56,7 +56,7 @@ private ArgumentConversionResult(Argument argument)
             if (argument.FirstParent?.Symbol is IdentifierSymbol identifierSymbol &&
                 argument.FirstParent.Next is null)
             {
-                var alias = identifierSymbol.Aliases.First();
+                var alias = identifierSymbol.GetLongestAlias(removePrefix: false);
```

Comment:
This wasn't really getting the longest alias but the first one.

Reply:
It was inconsistent. As you can see other places in the source code were using longest option alias to get its default alias. Here we were using the first one but it was a bug (at least in my opinion).

Reply:
Option.Name should not contain the prefix, while here we care about prefix.